activerecord/lib/active_record/model_schema.rb

```diff
@@ -126,6 +126,20 @@ module ModelSchema
     # +:immutable_string+. This setting does not affect the behavior of
     # <tt>attribute :foo, :string</tt>. Defaults to false.
 
+    ##
+    # :singleton-method: inheritance_column=
+    # :call-seq: inheritance_column=(column)
+    #
+    # Defines the name of the table column which will store the class name on single-table
+    # inheritance situations.
+    #
+    # The default inheritance column name is +type+, which means it's a
+    # reserved word inside Active Record. To be able to use single-table
+    # inheritance with another column name, or to use the column +type+ in
+    # your own model for something else, you can set +inheritance_column+:
+    #
+    #     self.inheritance_column = 'zoink'
```

Comment:
I split the documentation one part for `inheritance_column` and another for` inheritance_column= `

Reply:
I've got a quick question about the way we declare the `inheritance_column`.
It requires the use to the `self` keyword.
When comparing how we declare associations, the `has_many`, _belongs_to_ do not require `self` keyword.

```ruby
class Car < ApplicationRecord
  self.inheritance_column = 'car_type'

  has_many :parts
end
```

Is there a reason why we need to write it this ways`self.inheritance_column = 'car_type'` instead of `inheritance_column :car_type` ?

```ruby
class Car < ApplicationRecord
  inheritance_column :car_type

  has_many :parts
end
```
I'd like to understand the reason behind this.

Reply:
I'm not sure if I understand your question.

Are you asking why we need to write `self.`?  That is necessary because `inheritance_column` is implemented as a class attribute.  If we wrote just `inheritance_column = "foo"`, it would set a local variable `inheritance_column` instead of `MyClass.inheritance_column`.  (It's the same as with `attr_writer`.)

Or are you asking why `inheritance_column` was implemented as a class attribute instead of another way?  In that case, I do not have an objective answer.